lib/epoch.ex

```diff
@@ -4,32 +4,71 @@ defmodule DateTimeParser.Epoch do
   @token_key :unix_epoch
   @max_subsecond_digits 6
 
-  def parse(%{"seconds" => raw_seconds, "subseconds" => raw_subseconds}) do
-    with {seconds, ""} <- Integer.parse(raw_seconds) do
-      parsed_epoch =
-        case(raw_subseconds) do
-          "" ->
-            {seconds, {0, 0}}
+  def parse(%{"sign" => sign, "seconds" => raw_seconds, "subseconds" => raw_subseconds}) do
+    is_negative = if sign == "-", do: true, else: false
+    has_subseconds = if raw_subseconds == "", do: false, else: true
 
-          raw_subseconds ->
-            {subseconds, ""} = Float.parse("0.#{raw_subseconds}")
-            microseconds = (subseconds * :math.pow(10, 6)) |> trunc()
+    with {:ok, seconds} <- parse_seconds(raw_seconds, is_negative, has_subseconds),
+         {:ok, subseconds} <- parse_subseconds(raw_subseconds, is_negative) do
+      {:ok, [{@token_key, {seconds, subseconds}}], nil, nil, nil, nil}
+    end
+  end
 
-            truncated_microseconds =
-              microseconds
-              |> Integer.digits()
-              |> Enum.take(@max_subsecond_digits)
-              |> Integer.undigits()
+  @spec parse_seconds(String.t(), boolean(), boolean()) :: {:ok, integer()}
```

Comment:
Not sure how you feel about explicitly adding typespecs. I thought it would be nice to have the variable names listed to track what they are for as opposed to having to _know_ what the booleans meant. I also considered pattern matching on a map, but this seemed easier to start. Let me know what you prefer.

Reply:
I'm happy to have the typespecs. This is a private function internal to the library but the clarity helps. If you wanted to refactor to maps, that'd be good but up to you.

Reply:
Would you say using a `map` is what would be more common in other libraries? I also see a lot of keyword lists, but the required ordering seems a bit annoying. 

Now that it's done, I don't mind the explicit arguments as much, but I am open to change if it follows more common practices.

Reply:
Turning into maps may be a good refactoring for another PR. I have some thoughts on creating a Parser interface that might normalize the tokens into a map, and then `from_tokens` can pattern match from there.
I can't say maps are more common, but it's at least more convenient for pattern matching which I think can help make code simpler.

Reply:
> Turning into maps may be a good refactoring for another PR. I have some thoughts on creating a Parser interface that might normalize the tokens into a map, and then from_tokens can pattern match from there.

That sounds great!